// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package transcode_rpc;

// The greeting service definition.
service Transcode {
  rpc Setchannel (PIDS) returns (channel_list){}
  rpc Avio(avio_type) returns(StateReply){}
  rpc Process(empty) returns(StateReply){}
  rpc State(empty) returns(StateReply){} 
  rpc Stop(empty) returns(StateReply){}
}

// The request message containing the user's name.

message PIDS{
  int32 video=1;
  int32 audio=2;
  string filename=3;
}

message avio_type{
  int32 inputtype=1;// input frome memory = 0 from file = 1
  string file =2; //input from file name
}

// The response message containing the greetings
message StateReply {
  string reply = 1;
}
message channel_list {
  string list = 1;
}
message empty{

}